contracts/interfaces/IBaseWeightedMultisig.sol

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+interface IBaseWeightedMultisig {
+    error InvalidSigners();
+    error InvalidThreshold();
+    error MalformedSignatures();
+    error LowSignaturesWeight();
+    error InvalidWeights();
+
+    struct WeightedSigners {
```

Comment:
Since we're not exposing this in the API, this should be defined in the implementation instead